proof/proof.go

```diff
@@ -99,6 +99,16 @@ type Proof struct {
 	// NUMS key, to prove that the creator of the proof is able to produce
 	// a valid signature to spend the asset.
 	ChallengeWitness wire.TxWitness
+
+	// GenesisReveal is an optional Genesis struct, that must be provided
+	// for issuance proofs. This allows for derivation of the asset ID.
+	GenesisReveal *asset.Genesis
+
+	// GroupKeyReveal is the set of bytes that represent the internal key
+	// and Tapscript root used to derive the group key for the asset group.
+	// The field is defined as internal_key || tapscript_root, and must be
+	// provided for issuance proofs of grouped assets.
+	GroupKeyReveal *asset.GroupKeyReveal
```

Comment:
Add additional context here (and also the `GroupKeyReveal` struct) on the full derivation? So the first tweak with the asset ID info, then the extra layer with the actual tapscript root.

Reply:
Added as godoc for the actual `GroupKeyReveal` struct.